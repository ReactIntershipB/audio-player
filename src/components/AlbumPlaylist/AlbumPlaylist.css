.playlist-container {
    max-width: 1000px;
    margin: 20px auto 130px auto;
    padding: 0 20px;
    text-align: left;
}

.avatar {
    display: flex;
    align-items: center;
    margin-bottom: 40px;
}

.avatar > :first-child {
    margin-right: 20px;
}

/*from https://css-tricks.com/examples/hrs/*/
hr {
    border: 0;
    height: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(255, 255, 255, 0.3);
}

.ant-spin-spinning.spinner {
    position: absolute;
    top: 50%;
    left: 50%;
}