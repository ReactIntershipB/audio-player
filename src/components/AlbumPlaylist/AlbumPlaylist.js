import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { observer, inject } from 'mobx-react';
import { Alert } from 'antd';

import Spinner from './../common/Spinner/Spinner';
import { ListComponent } from '../common/ListComponent/ListComponent';
import { MessageBox } from '../common/MessageBox';

@inject('albumModel')
@observer
export default class AlbumPlaylist extends Component {
  componentDidMount() {
    this.props.albumModel.find(this.props.match.params.id);
  }

  get playlist() {
    const { data, loading } = this.props.albumModel;

    return (loading || data.error || !data.tracks ? null : <ListComponent heading={data.title} data={data.tracks.data} type={'track'} albumTitle={data.title} />);
  }

  get spinner() {
    const { loading } = this.props.albumModel;
    return loading ? <Spinner /> : null;
  }

  get errorMessage() {
    return this.props.albumModel.isError
    ? <MessageBox>
      <Alert message="Something went wrong with download data. Please try to refresh the page." type="error" showIcon />
    </MessageBox>
    : null;
  }

  render() {
    return (
      <div className="component-container">
        {this.spinner}
        {this.playlist}
        {this.errorMessage}
      </div>
    );
  }
}

AlbumPlaylist.propTypes = {
  mediator: PropTypes.object,
  albumModel: PropTypes.object,
  match: PropTypes.any
};
