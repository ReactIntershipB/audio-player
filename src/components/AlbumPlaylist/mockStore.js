import { action, observable } from 'mobx';

const mockData = {
    id: 302127,
    title: 'Discovery',
    cover_small: 'https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/56x56-000000-80-0-0.jpg',
    tracks: {
        data: [
            {
                id: 3135553,
                title: 'One More Time',
                duration: 320,
                artist: {
                    name: 'Daft Punk'
                }
            }
        ]
    }
};

export class AlbumModel {
    @observable data = [];
    @observable loading = false;

    @action
    find = () => {
        this.getData();
    }

    getData = () => {
        this.data = mockData;
    }
}

export const albumModel = new AlbumModel();
export const songModel = {};
export const playerModel = {};
