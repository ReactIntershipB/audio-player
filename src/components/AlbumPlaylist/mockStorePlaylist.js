import { observable } from 'mobx';

export const mockData = {
    id: 302127,
    title: 'Discovery',
    cover_small: 'https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/56x56-000000-80-0-0.jpg',
    tracks: {
        data: [
            {
                id: 3135553,
                title: 'One More Time',
                duration: 320,
                artist: {
                    name: 'Daft Punk'
                }
            }
        ]
    }
};

export class AlbumModel {
    @observable data = [];
    @observable loading = false;

    find = jest.fn(() => {
        this.getData();
    });

    getData = () => {
        this.data = mockData;
    }
}
