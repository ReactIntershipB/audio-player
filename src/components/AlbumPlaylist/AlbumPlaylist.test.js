import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';

import AlbumPlaylist from './AlbumPlaylist';
import * as mockStore from './mockStorePlaylist';

describe('AlbumPlaylist', () => {
  const match = {
    params: {
      id: 123456
    }
  };

  it('should match snapshot', () => {
    const props = {
      albumModel: new mockStore.AlbumModel(),
      songModel: {},
      playerModel: {},
      appUI: {}
    };

    const albumPlaylist = renderer
      .create(
        <Provider
          albumModel={props.albumModel}
          songModel={props.songModel}
          playerModel={props.playerModel}
          appUI={props.appUI}
        >
          <AlbumPlaylist match={match} />
        </Provider>
        ).toJSON();

    expect(albumPlaylist).toMatchSnapshot();
  });

  it('should call the find function with proper arguments', () => {
    const props = {
      albumModel: new mockStore.AlbumModel(),
      songModel: {},
      playerModel: {},
      appUI: {}
    };
    const find = props.albumModel.find;

    renderer
      .create(
        <Provider
          albumModel={props.albumModel}
          songModel={props.songModel}
          playerModel={props.playerModel}
          appUI={props.appUI}
        >
          <AlbumPlaylist match={match} />
        </Provider>
      );

    expect(find).toHaveBeenCalled();
    expect(find).toHaveBeenCalledWith(match.params.id);
    expect(find).toHaveBeenCalledTimes(1);
  });
});
