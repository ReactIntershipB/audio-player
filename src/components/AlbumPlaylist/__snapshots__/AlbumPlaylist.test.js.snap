// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumPlaylist should match snapshot 1`] = `
<div>
  <div
    className="playlist-container"
  >
    <div
      className="avatar"
    >
      <span
        className="ant-avatar ant-avatar-square ant-avatar-image"
        style={
          Object {
            "fontSize": 18,
            "height": 64,
            "lineHeight": "64px",
            "width": 64,
          }
        }
      >
        <img
          alt={undefined}
          onError={[Function]}
          src="https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/56x56-000000-80-0-0.jpg"
        />
      </span>
      <h2>
        Discovery
      </h2>
    </div>
    <div>
      <div>
        <div
          className="ant-list-item"
        >
          <div
            className="ant-list-item-meta"
          >
            <div
              className="ant-list-item-meta-avatar"
            >
              <button
                className="ant-btn ant-btn-circle ant-btn-lg ant-btn-icon-only"
                data-testid="play-button"
                disabled={undefined}
                onClick={[Function]}
                type="button"
              >
                <i
                  className="anticon anticon-caret-right"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="caret-right"
                    fill="currentColor"
                    height="1em"
                    onClick={undefined}
                    style={undefined}
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M392.7 76.3l489.1 403.1a42.53 42.53 0 0 1 0 63.4L392.7 947.7c-27.4 24.4-70.8 5-70.8-31.7V107.9c.1-36.6 43.4-56.1 70.8-31.6z"
                    />
                  </svg>
                </i>
              </button>
            </div>
            <div
              className="ant-list-item-meta-content"
            >
              <h4
                className="ant-list-item-meta-title"
              >
                One More Time
              </h4>
              <div
                className="ant-list-item-meta-description"
              >
                Daft Punk - One More Time
              </div>
            </div>
          </div>
          <div
            className="ant-list-item-content"
          >
            <div>
              5:20
            </div>
          </div>
        </div>
        <hr />
      </div>
    </div>
  </div>
</div>
`;
